### PR TITLE
ViT-B, ViT-L models

### PR DESCRIPTION
**Pull request by abhiskk**
Test plan:

```python
import torch

from diht import model_zoo
from PIL import Image


text_tokenizer, image_transform, model = model_zoo.load_model(
    "diht_vitl14_336px", is_train=False
)

image = Image.open("infer_image.png").convert("RGB")
image = image_transform(image).unsqueeze(0)
text_captions = ["a mountain", "a beach", "a desert"]
text = text_tokenizer(text_captions)

with torch.no_grad():
    image_features, text_features, logit_scale = model(image, text)
    logits_per_image = logit_scale * image_features @ text_features.T
    probs = logits_per_image.softmax(dim=-1).numpy()

print(f"text captions: {text_captions}")
print(f"text caption probs: {probs}")
```

### CODE REVIEW

**Comment by abhimanyudubey** [comment]
lgtm, do we not need the SHA checksum?

File: diht/model.py [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+        x = self.ln_post(x[:, 0, :])
+
+        if self.proj is not None:
+            x = x @ self.proj
+
+        return x
+
+
+@dataclass
+class CLIPVisionCfg:
```
Comment (filipradenovic) [inline diht/model.py]: DiHTVisionCfg?

File: diht/model.py [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+    head_width: int = 64
+    mlp_ratio: float = 4.0
+    patch_size: int = 16
+    image_size: Union[Tuple[int, int], int] = 224
+    image_resize_res: Union[Tuple[int, int], int] = None
+    project_vision_fts: bool = False
+
+
+@dataclass
+class CLIPTextCfg:
```
Comment (filipradenovic) [inline diht/model.py]: DiHTTextCfg?

File: tests/test_model.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+# import numpy as np
```
Comment (filipradenovic) [inline tests/test_model.py]: Why is this whole file commented out?

File: diht/model.py [hunk 4]
```diff
@@ -0,0 +1,454 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+
+
```
Comment (SimonVandenhende) [inline diht/model.py:5]: Was this code referenced from another repository? Lets add a link.

File: diht/model_zoo.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+import logging
+import os
+import urllib
+from pathlib import Path
```
Comment (SimonVandenhende) [inline diht/model_zoo.py]: I think code in this file might have been referenced from somewhere else. Lets put appropriate link if so.

File: tests/test_model.py (hunk 3 above)
Comment (abhiskk) [reply tests/test_model.py]: added tests now, they were not present when PR was opened.

**Review by dkm2110 — APPROVED** [review]

**Pull request merged**